sync/pooled_worker_pool.go

```diff
@@ -110,25 +112,36 @@ func (p *pooledWorkerPool) Go(work Work) {
 		// will slowly shrink the pool back down to its original size because
 		// workers created in this manner will not spawn their replacement
 		// before killing themselves.
-		p.spawnWorker(work, workCh, false)
+		p.spawnWorker(uint64(currTime), work, workCh, false)
 	}
 }
 
 func (p *pooledWorkerPool) spawnWorker(
-	initialWork Work, workCh chan Work, spawnReplacement bool) {
+	seed uint64, initialWork Work, workCh chan Work, spawnReplacement bool) {
 	go func() {
 		p.incNumRoutines()
 		if initialWork != nil {
 			initialWork()
 		}
 
 		// RNG per worker to avoid synchronization.
-		rng := rand.New(rand.NewSource(p.nowFn().UnixNano()))
+		var (
+			rng = pcg.NewPCG64().Seed(seed, seed*2, seed*3, seed*4)
```

Comment:
can you check this distribution empirically once

Reply:
did, it was good